### PR TITLE
Tune the behavior of the Paratime Picker - Clicky version

### PR DESCRIPTION
**Pull request by csillag**
Rationale coming at the next chance to talk.

See #479  for a no-click alternative.

### CODE REVIEW

**Comment by csillag** [comment]
After discussing this with @donouwens , we came to the conclusion that this is not (necessarily) what we want to do here.
Let's come back to this after I have created a new test version based on what we have discussed.

**Comment by csillag** [comment]
Updated as discussed

File: src/app/components/ParaTimePicker/index.tsx [hunk 1]
```diff
@@ -36,15 +38,26 @@ const ParaTimePickerContent: FC<ParaTimePickerContentProps> = ({ onClose, onConf
...
   const [showNetworkMenu, setShowNetworkMenu] = useState(network !== Network.mainnet)
-  const [selectedLayer, setSelectedLayer] = useState<undefined | Layer>()
-  const [selectedNetwork, setSelectedNetwork] = useState<undefined | Network>(network)
-  const [hoveredLayer, setHoveredLayer] = useState<undefined | Layer>()
-  const [hoveredNetwork, setHoveredNetwork] = useState<undefined | Network>()
+  const [selectedLayer, setSelectedLayer] = useState<Layer>(layer)
+  const [selectedNetwork, setSelectedNetwork] = useState<Network>(network)
+  const selectNetwork = (newNetwork: Network) => {
+    setSelectedNetwork(newNetwork)
+    setSelectedLayer(RouteUtils.getEnabledLayersForNetwork(network)[0])
```
Comment (lukaw3d) [inline src/app/components/ParaTimePicker/index.tsx:45]: - https://683bae47.oasis-explorer.pages.dev/mainnet/emerald
- switch to testnet
- shows "Emerald Testnet"
- click Select
- crashes, unsupported_layer

Fix is probably just
```suggestion
    setSelectedLayer(RouteUtils.getEnabledLayersForNetwork(newNetwork)[0])
```

File: src/app/components/ParaTimePicker/LayerMenu.tsx [hunk 2]
```diff
@@ -54,12 +57,14 @@ export const LayerMenuItem: FC<LayerMenuItemProps> = ({
...
       divider={divider}
       onMouseEnter={() => {
-        if (layer !== selectedLayer) {
-          setSelectedLayer(undefined)
-        }
         setHoveredLayer(layer)
       }}
-      onClick={() => setSelectedLayer(layer)}
+      onMouseLeave={() => {
+        setHoveredLayer()
```
Comment (lukaw3d) [inline src/app/components/ParaTimePicker/LayerMenu.tsx:63]: We can later simplify to css :hover

File: src/app/components/ParaTimePicker/index.tsx [hunk 3]
```diff
@@ -57,62 +70,57 @@ const ParaTimePickerContent: FC<ParaTimePickerContentProps> = ({ onClose, onConf
...
-          </Button>
-          <Button
-            onClick={() => onConfirm(selectedNetwork!, selectedLayer!)}
-            disabled={!selectedLayer || !selectedNetwork}
-            color="primary"
-            variant="contained"
-          >
-            {t('common.select')}
-          </Button>
+          {selectedLayer && (
```
Comment (lukaw3d) [inline src/app/components/ParaTimePicker/index.tsx:104]: `selectedLayer && ` Probably no longer needed

File: src/app/components/ParaTimePicker/index.tsx [hunk 4]
```diff
@@ -57,62 +70,57 @@ const ParaTimePickerContent: FC<ParaTimePickerContentProps> = ({ onClose, onConf
...
+              color="secondary"
+              variant="outlined"
+              sx={{ textTransform: 'capitalize' }}
+            >
+              {t('common.cancel')}
+            </Button>
+          )}
+          {selectedLayer && (
+            <Button
+              onClick={() => onConfirm(selectedNetwork!, selectedLayer!)}
```
Comment (lukaw3d) [inline src/app/components/ParaTimePicker/index.tsx:116]:
```suggestion
              onClick={() => onConfirm(selectedNetwork, selectedLayer)}
```

**Review by lukaw3d — APPROVED** [review]
yay

File: src/app/components/ParaTimePicker/index.tsx (hunk 1 above)
Comment (csillag) [reply src/app/components/ParaTimePicker/index.tsx]: Yes that was exactly the original intention.

File: src/app/components/ParaTimePicker/index.tsx (hunk 3 above)
Comment (csillag) [reply src/app/components/ParaTimePicker/index.tsx]: Yes, removed

File: src/app/components/ParaTimePicker/index.tsx (hunk 4 above)
Comment (csillag) [reply src/app/components/ParaTimePicker/index.tsx]: Yep.

File: src/app/components/ParaTimePicker/LayerMenu.tsx (hunk 2 above)
Comment (csillag) [reply src/app/components/ParaTimePicker/LayerMenu.tsx:63]: Agreed.

**Pull request merged**